File: legacy_python/app/models/learning.py
import mongoengine as db

class Subject(db.Document):
    name = db.StringField(required=True, max_length=100)
    description = db.StringField(max_length=255)
    icon = db.StringField(default='fa-book', max_length=50)

class Chapter(db.Document):
    subject = db.ReferenceField(Subject, required=True)
    title = db.StringField(required=True, max_length=150)
    order = db.IntField(default=0)

class Lesson(db.Document):
    chapter = db.ReferenceField(Chapter, required=True)
    title = db.StringField(required=True, max_length=150)
    content = db.StringField() # Markdown or HTML content
    video_url = db.StringField(max_length=255)

class Material(db.Document):
    lesson = db.ReferenceField(Lesson, required=True)
    title = db.StringField(required=True, max_length=150)
    file_path = db.StringField(required=True, max_length=255)
    file_type = db.StringField(max_length=20)

class LearningPath(db.Document):
    title = db.StringField(required=True, max_length=150)
    description = db.StringField(max_length=255)
    student = db.ReferenceField('Student') # Optional specific path for student

File: src/models/Learning.js
const mongoose = require('mongoose');

const subjectSchema = new mongoose.Schema({
    name: { type: String, required: true, maxlength: 100 },
    description: { type: String, maxlength: 255 },
    icon: { type: String, default: 'fa-book', maxlength: 50 }
});

const chapterSchema = new mongoose.Schema({
    subject: { type: mongoose.Schema.Types.ObjectId, ref: 'Subject', required: true },
    title: { type: String, required: true, maxlength: 150 },
    order: { type: Number, default: 0 }
});

const lessonSchema = new mongoose.Schema({
    chapter: { type: mongoose.Schema.Types.ObjectId, ref: 'Chapter', required: true },
    title: { type: String, required: true, maxlength: 150 },
    content: { type: String }, // Markdown or HTML
    video_url: { type: String, maxlength: 255 }
});

const materialSchema = new mongoose.Schema({
    lesson: { type: mongoose.Schema.Types.ObjectId, ref: 'Lesson', required: true },
    title: { type: String, required: true, maxlength: 150 },
    file_path: { type: String, required: true, maxlength: 255 },
    file_type: { type: String, maxlength: 20 }
});

const learningPathSchema = new mongoose.Schema({
    title: { type: String, required: true, maxlength: 150 },
    description: { type: String, maxlength: 255 },
    student: { type: mongoose.Schema.Types.ObjectId, ref: 'Student' } // Optional
});

module.exports = {
    Subject: mongoose.model('Subject', subjectSchema),
    Chapter: mongoose.model('Chapter', chapterSchema),
    Lesson: mongoose.model('Lesson', lessonSchema),
    Material: mongoose.model('Material', materialSchema),
    LearningPath: mongoose.model('LearningPath', learningPathSchema)
};
